nuget/lib/dependabot/nuget/file_fetcher.rb

```diff
@@ -124,36 +122,29 @@ def fetch_directory_build_files
         attempted_dirs = []
         directory_build_files = []
 
-        # Don't need to insert "." here, because Directory.Build.props files
-        # can only be used by project files (not packages.config ones)
+        # find all build files (Directory.Build.props/.targets) relative to the given project file
         project_files.map { |f| File.dirname(f.name) }.uniq.each do |dir|
-          possible_dirs = dir.split("/").flat_map.with_index do |_, i|
-            dir.split("/").first(i).join("/")
-          end.reverse
+          # Simulate MSBuild walking up the directory structure looking for a file
+          possible_dirs = dir.split("/").map.with_index do |_, i|
+            dir.split("/").first(i + 1).join("/")
+          end.reverse + ["/"]
 
           possible_dirs.each do |possible_dir|
             break if attempted_dirs.include?(possible_dir)
 
             attempted_dirs << possible_dir
-            file = repo_contents(dir: possible_dir).find { |f| possible_build_file_paths.include?(f.name) }
-            directory_build_files << fetch_file_from_host(File.join(possible_dir, file.name)) if file
+            build_files = repo_contents(dir: possible_dir).select { |f| f.name.match?(BUILD_FILE_NAMES) }
+            directory_build_files += build_files.map do |file|
+              possible_file = File.join(possible_dir, file.name)
+              possible_file = possible_file[1..-1] if possible_file.start_with?("/") # remove leading slash
```

Comment:
`possible_file.delete_prefix("/")` is probably a bit more clear, and it self explains so you can remove the comment.

Reply:
My lack of Ruby experience is showing :)